src/language/dafnyInstallation.ts

```diff
@@ -65,14 +81,28 @@ export class DafnyInstaller {
     private readonly statusOutput: OutputChannel
   ) {}
 
-  public static isMinimumRequiredLanguageServer(version: string): boolean {
+  public isLatestKnownLanguageServerOrNewer(version: string): boolean {
     if(version === LanguageServerConstants.UnknownVersion) {
+      this.writeStatus('failed to resolve the installed Dafny version');
       return true;
     }
-    const [ givenMajor, givenMinor ] = version.split('.');
-    const [ requiredMajor, requiredMinor ] = LanguageServerConstants.RequiredVersion.split('.');
-    return givenMajor > requiredMajor
-      || givenMajor === requiredMajor && givenMinor >= requiredMinor;
+    const givenParts = version.split('.');
+    const latestVersion = LanguageServerConstants.LatestVersion;
+    const latestParts = latestVersion.split('.');
+    for(let i = 0; i < Math.min(givenParts.length, latestParts.length); i++) {
+      const given = givenParts[i];
+      const latest = latestParts[i];
+      if(given < latest) {
+        this.writeStatus(`the installed Dafny version is older than the latest: ${version} < ${latestVersion}`);
+        return false;
+      }
+      if(given > latest) {
+        this.writeStatus(`the installed Dafny version is newer than the latest: ${version} > ${latestVersion}`);
+        return true;
+      }
+    }
+    this.writeStatus(`the installed Dafny version is the latest known: ${version} = ${latestVersion}`);
```

Comment:
This will report `=` for `3.2` vs `3.2.1`, won't it?

Reply:
Exactly. I've taken the minimum of both lengths to account for cases where the full version string is not available for one or another.
Moreover, it's relevant to account for the build-version that is provided by the language server:
```
the installed Dafny version is the latest known: 3.3.0.31104 = 3.3.0
```